mlflow/server/js/src/components/artifact-view-components/ShowArtifactImageView.js

```diff
@@ -52,61 +67,72 @@ class ShowArtifactImageView extends Component {
+        />
+      </React.Fragment>
+    );
+  };
+
+  renderStaticImage = () => {
     const { loading, dataURL, width, height } = this.state;
 
     if (loading) return <div>Loading...</div>;
 
+    return (
+      <Plot
+        layout={{
+          width: width * (500 / height),
+          height: 500,
+          xaxis: { visible: false, range: [0, width] },
+          yaxis: { visible: false, range: [0, height], scaleanchor: 'x', scaleratio: 1 },
+          images: [
+            {
+              source: dataURL,
+              xref: 'x',
+              yref: 'y',
+              x: 0,
+              y: 0,
+              xanchor: 'left',
+              yanchor: 'bottom',
+              sizex: width,
+              sizey: height,
+            },
+          ],
+          margin: { l: 0, r: 0, t: 0, b: 0 },
+        }}
+        config={{
+          displaylogo: false,
+          scrollZoom: true,
+          doubleClick: 'reset',
+          modeBarButtonsToRemove: [
+            'hoverCompareCartesian',
+            'hoverClosestCartesian',
+            'lasso2d',
+            'sendDataToCloud',
+            'select2d',
+            'toggleSpikelines',
+          ],
+        }}
+        useResizeHandler
+      />
+    );
+  };
+
+  render() {
     return (
       <div className="image-outer-container">
         <div className="image-container">
-          {
-            <Plot
-              data={[
-                {
-                  x: [0, width],
-                  y: [0, height],
-                  type: 'scatter',
-                  mode: 'markers',
-                  marker: { opacity: 0, size: 0 },
-                  hoverinfo: 'none',
-                },
-              ]}
-              layout={{
-                width: width * (500 / height),
-                height: 500,
-                xaxis: { visible: false, autorange: true },
-                yaxis: { visible: false, autorange: true, scaleanchor: 'x', scaleratio: 1 },
-                images: [
-                  {
-                    source: dataURL,
-                    xref: 'x',
-                    yref: 'y',
-                    x: 0,
-                    y: 0,
-                    xanchor: 'left',
-                    yanchor: 'bottom',
-                    sizex: width,
-                    sizey: height,
-                  },
-                ],
-                margin: { l: 0, r: 0, t: 0, b: 0 },
-              }}
-              config={{
-                displaylogo: false,
-                scrollZoom: true,
-                modeBarButtonsToRemove: [
-                  'hoverCompareCartesian',
-                  'hoverClosestCartesian',
-                  'lasso2d',
-                  'sendDataToCloud',
-                  'select2d',
-                  'toggleSpikelines',
-                ],
-              }}
-              useResizeHandler
-            />
-          }
+          {this.isGif() ? this.renderGif() : this.renderStaticImage()}
```

Comment:
One last request, would you mind adding a test for ShowArtifactImageView? Just basic rendering tests plus a test that mocks `renderGif` and `renderStaticImage` and make sure the correct one is call according to the file type?

Reply:
ok I'll add it.